spec/controllers/api/v1/punches_controller_spec.rb

```diff
@@ -0,0 +1,155 @@
+require 'spec_helper'
+
+describe Api::V1::PunchesController, :type => :controller do
+  let(:company) { create(:company) }
+  let(:user) { create(:user, :with_token, company: company) }
+  let(:project) { create(:project, company: user.company) }
+  let(:headers) { { token: user.token } }
+
+  describe 'POST api/v1/punches' do
+    context 'when api token is valid and params are valid' do
+      let(:params) {
+        {
+          "_json": [
```

Comment:
Instead of calling this main key `_json`, you could call it `punches` for better readability.